File: Meals_Android/app/src/main/java/org/techtown/project5/Activity/MainActivity.java
package org.techtown.project5.Activity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;

import org.techtown.project5.R;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // SharedPreferences 속성
        SharedPreferences sf = getSharedPreferences("sFile", MODE_PRIVATE);

        // check 변수에 getInt 속성을 사용하여 SchoolActivity 클래스에서 check 라는 이름으로 저장한 값을 불러옵니다.
        int check = sf.getInt("check",0);

        // check 변수의 값이 1이라면 사용자는 이미 애플리케이션을 사용한 적이 있습니다.
        if(check == 1){

           Intent checkMeals = new Intent(MainActivity.this, MealsActivity.class);
           startActivity(checkMeals);

        }

        // check 변수의 값이 0 이라면 사용자는 애플리케이션을 이번에 처음으로 사용하는 것입니다.
        else if(check == 0){

            Intent searchSchool = new Intent(MainActivity.this, SchoolActivity.class);

            startActivity(searchSchool);
        }


    }




}

File: Meals_Android V2/app/src/main/java/com/project/meals/view/MainActivity.kt
package com.project.meals.view

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.project.meals.R
import com.project.meals.databinding.ActivityMainBinding
import com.project.meals.view.base.BaseActivity
import com.project.meals.view.intro.AppIntroActivity
import com.project.meals.view.meals.MealsActivity
import com.project.meals.viewmodel.MainViewModel

class MainActivity : BaseActivity() {

    lateinit var binding : ActivityMainBinding
    lateinit var viewModel : MainViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // SharedPreferences
        var checkLogin : SharedPreferences = getSharedPreferences("checkLogin", Context.MODE_PRIVATE)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        viewModel = ViewModelProviders.of(this@MainActivity).get(MainViewModel::class.java)

        binding.viewModel = viewModel
        binding.lifecycleOwner

        observerViewModel()
        viewModel.CheckLogin(checkLogin)
    }

    fun observerViewModel(){
        with(viewModel){
            onSuccessEvent.observe(this@MainActivity, Observer {
                startActivity(Intent(this@MainActivity, MealsActivity::class.java))
                finish()
            })
            onFailEvent.observe(this@MainActivity, Observer {
                startActivity(Intent(this@MainActivity, AppIntroActivity::class.java))
                finish()
            })
        }
    }
}
